src/Sylius/Bundle/OrderBundle/DependencyInjection/Compiler/RegisterAdjustmentFactoryPass.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\OrderBundle\DependencyInjection\Compiler;
+
+use Sylius\Component\Attribute\Factory\AttributeFactory;
+use Sylius\Component\Order\Factory\AdjustmentFactory;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+class RegisterAdjustmentFactoryPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $oldAdjustmentFactory = $container->getDefinition('sylius.factory.adjustment');
```

Comment:
I don't think we need a compiler pass here at all. All services used are already available in the container. I'd suggest using - http://symfony.com/doc/current/components/dependency_injection/advanced.html#decorating-services